### PR TITLE
Add plugins documentation

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

The recently introduced `warren indicator` CLI command supposes plugins to be properly declared. We need to start documenting API plugins guidelines.

## Proposal

- [x] format documentation using prettier
- [x] enforce documentation sources linting in the CI with prettier
- [x] document API plugins requirements

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -273,7 +273,8 @@ lint: ## lint api, app and frontend sources
 lint: \
   lint-api \
   lint-app \
-  lint-frontend
+  lint-frontend \
+	lint-docs
```
Comment (jmaupetit) [inline Makefile:277]: outch

**Review by wilbrdt — COMMENTED** [review]
Wow, great first draft! :tada:

File: docs/index.md
Comment (quitterie-lcs) [inline docs/index.md:1]: Can you remove the absolute link (/) for Warren index docs page (line 14)?

**Review by quitterie-lcs — APPROVED** [review]

File: docs/index.md
Comment (jmaupetit) [reply docs/index.md:1]: My bad! I've switched to a relative link. Should be ok now.

**Pull request merged**